src/emulate.c

```diff
@@ -135,7 +188,8 @@ RV_EXCEPTION_LIST
         rv->compressed = compress;                                    \
         rv->csr_cycle = cycle;                                        \
         rv->PC = PC;                                                  \
-        rv_except_##type##_misaligned(rv, IIF(IO)(addr, mask_or_pc)); \
+        rv->is_trapped = true;                                        \
+        rv_trap_##type##_misaligned(rv, IIF(IO)(addr, mask_or_pc));   \
```

Comment:
Shall we regard misalignment as a special case? Or, can it be generalized?

Reply:
May I know what does the special case refer to?

Reply:
> May I know what does the special case refer to?

I meant to ask whether misalignment exceptions should be treated as one of the supported traps. If so, checks for misalignment exceptions should be implemented indirectly.

Reply:
According to the [entry.S](https://elixir.bootlin.com/linux/v6.1.112/source/arch/riscv/kernel/entry.S#L582) of Linux v6.1.112, it does support `do_trap_load_misaligned` and `do_trap_store_misaligned`. Thus, any misaligned load/store fault should be trapped into same path. In other words, `trap_handler` should be capable to handle the trap.